search/constraint.go

```diff
@@ -292,7 +272,7 @@ func (ec *equalConstraint) filter(ctx context.Context, rg parquet.RowGroup, prim
 		default:
 			off, count := bl, 0
 			for j := bl; j < br; j++ {
-				if ec.comp(ec.val, symbols.Get(j)) != 0 {
+				if !ec.matches(symbols.Get(j)) {
```

Comment:
If it is null value, we can get the row we need from `pg.DefinitionLevels()`. It should be more performant than matching symbols here, no?